### ISSUE TITLE
Provide a means to calculate the relative directory of input paths

### ISSUE THREAD

**Issue by samuell** [common-workflow-language/common-workflow-language#213]
*TL;DR: Suggestion: Add an "input directory" variable to runtime environment (e.g. `runtime.indir`)  or a "relative path" to input objects (e.g. `inputs.myinput.relpath`).*

When calculating output paths that are based on an input's path, there is currently a problem: The input paths are full absolute paths, and there is no way (AFAIS) to compute the relative path based on the available information in the runtime environment.

As an example, say that we have a workflow with an input named `inp`, and an output named `outp`, where we want to give the filename on outp the same as the file on inp, but with a suffix `.my_suffix` added. Then, we might do something like this:

```yaml
#!/usr/bin/env cwl-runner
cwlVersion: "cwl:draft-3"
class: CommandLineTool

requirements:
  - $import: envvar-global.yml
  - class: InlineJavascriptRequirement

inputs:
  - id: "inp"
    type: File
    inputBinding:
      position: 3

outputs:
  - id: outp
    type: File
    outputBinding:
      glob: ${ inputs.inp.path + ".my_suffix" }
```

The first thing we can notice is that 

```javascript
inputs.inp.path + ".my_suffix"
```

will get problematic, since `inputs.inp.path` is an absolute URL (say for example `~/work/cwlruns/data`, which is not allowed in the code, and most importantly, will not work when the output folder is `/tmp/[some-id]` ... since the full path will be something else, and so glob will not find the output file.

So, we might change this line to something like this (as [Nebojsa suggsted](https://groups.google.com/d/msg/common-workflow-language/nxgfP6u0Ku4/FRIBBGuJIgAJ)), that extracts the filename only, omitting the "full folder path" part of the path:

```javascript
inputs.inp.path.split("/").pop() + ".my_suffix"
```

The problem here though, is that if we have our files in a path `data/inputfile.txt`, relative to our workflow scripts, this will not work, since it extracts only the filename.

Thus, I think one needs either of these two information bits:

1. A variable `indir` (input directory), which contains the full folder path to where the input data is located. This could be used to remove the full folder path from the "full path" input string:

    ```javascript
    inputs.inp.path.replace(runtime.indir+"/", "") + ".my_suffix"
    ```

2. Even better, would be to have a variable `relpath` (relative path), directly on the input object, which would contain a relative path to the input data file. Then this one could be used directly, to construct output paths:

    ```javascript
    inputs.inp.relpath + ".my_suffix"
    ```

Note that using the existing `runtime.outdir`, like so (as [was suggested in this thread](https://groups.google.com/d/msg/common-workflow-language/DDIyEyWReR0/SbfQnFLAIQAJ)):

```javascript
inputs.inp.path.replace(runtime.outdir+"/", "") + ".my_suffix"
```

... does not work when runtime.outdir is a path of the form `/tmp/[some-id]` (remember, the input path was on the form `~/work/cwlruns/data`).

**Comment by samuell** [comment]
Ah, just learned now that the whole idea with the `/tmp/` directories, are for enabling a mount point for docker ... and that the problems we've ran into are mostly because of [cwltool / issue 80](https://github.com/common-workflow-language/cwltool/issues/80), that stuff is trying to be run in docker even though the docker requirement is just set in `hints:`.

When either removing the `hints: ...docker...` part completely, or running properly through docker, things work out fine.

But this also means an even higher prio to get this (issue #213) fixes asap...

**Comment by mr-c** [comment]
Thank you @samuell for this proposal.

FYI: The specification does not currently require that input files are in the same directory, and the reference implementation does not do so. 

The [InitialWorkDirRequirement](http://www.commonwl.org/v1.0/CommandLineTool.html#InitialWorkDirRequirement) added in `v1.0` should be sufficient for your needs. Please let us know if that doesn't work or is too awkward.
